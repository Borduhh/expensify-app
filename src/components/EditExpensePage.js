import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';

import { startEditExpense, startRemoveExpense } from '../actions/expenses';

export class EditExpensePage extends React.Component {
  handleOnSubmit = (expense) => {
    this.props.startEditExpense(this.props.expense.id, expense);
    this.props.history.push('/');
  };

  handleOnClick = () => {
    this.props.startRemoveExpense({ id: this.props.expense.id });
    this.props.history.push('/');
  };

  render() {
    return (
      <div>
        <div className='page-header'>
          <div className='content-container'>
            <h2 className='page-header__title'>Edit Expense</h2>
          </div>
        </div>
        <div className='content-container'>
          <ExpenseForm
            expense={this.props.expense}
            onSubmit={this.handleOnSubmit} />
          <button className='button--secondary-full-width' onClick={this.handleOnClick}>Remove Expense</button>
        </div>
      </div>
    );
  };
}

const mapStateToProps = (state, props) => {
  return {
    expense: state.expenses.find((expense) => expense.id === props.match.params.id)
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    startEditExpense: (id, expense) => { dispatch(startEditExpense(id, expense)) },
    startRemoveExpense: (expenseId) => { dispatch(startRemoveExpense(expenseId)) }
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);