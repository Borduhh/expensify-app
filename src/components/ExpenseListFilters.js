import React from 'react';
import { connect } from 'react-redux'
import { DateRangePicker } from 'react-dates';

import { setTextFilter, sortByDate, sortByAmount, setStartDate, setEndDate } from '../actions/filters';

export class ExpenseListFilters extends React.Component {

  state = {
    calendarFocused: null
  };

  onDatesChange = ({ startDate, endDate }) => {
    this.props.setStartDate(startDate);
    this.props.setEndDate(endDate);
  };

  onFocusChange = (calendarFocused) => {
    this.setState(() => ({ calendarFocused }))
  };

  onTextChange = (e) => {
    this.props.setTextFilter(e.target.value);
  };

  onSortByChange = (e) => {
    if (e.target.value === 'date') {
      this.props.sortByDate();
    } else if (e.target.value === 'amount') {
      this.props.sortByAmount();
    }
  }

  render() {
    return (
      <div className='input-group content-container'>
        <div className='input-group__item input-group__item--text-filter'>
          <input
            type='text'
            className='text-input'
            placeholder='Search Expenses...'
            value={
              this.props.filters.text
            }
            onChange={this.onTextChange}
          />
        </div>
        <div className='input-group__item input-group__item--sort-filter'>
          <select
            className='select'
            value={
              this.props.filters.sortBy
            }
            onChange={this.onSortByChange}
          >
            <option value='date'> Date </option>
            <option value='amount' > Amount </option>
          </select>
        </div>
        <div className='input-group__item input-group__item--date-filter'>
          <DateRangePicker
            startDate={this.props.filters.startDate}
            startDateId='filterDatesStartDate'
            endDate={this.props.filters.endDate}
            endDateId='filterDatesEndDate'
            onDatesChange={this.onDatesChange}
            focusedInput={this.state.calendarFocused}
            onFocusChange={this.onFocusChange}
            numberOfMonths={1}
            isOutsideRange={() => false}
            showClearDates={true}
          />
        </div>
      </div>
    );
  };
}

const mapStateToProps = (state) => {
  return {
    filters: state.filters
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    setStartDate: (startDate) => dispatch(setStartDate(startDate)),
    setEndDate: (endDate) => dispatch(setEndDate(endDate)),
    setTextFilter: (text) => dispatch(setTextFilter(text)),
    sortByDate: () => dispatch(sortByDate()),
    sortByAmount: () => dispatch(sortByAmount())
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilters);