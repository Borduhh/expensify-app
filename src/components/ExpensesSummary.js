import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import { Link } from 'react-router-dom';

import { getVisibleExpenses } from '../selectors/expenses';
import { getExpensesTotal } from '../selectors/expenses-total';

export const ExpensesSummary = ({expensesCount, expensesTotal}) => {
  const expenseWord = expensesCount === 1 ? 'expense' : 'expenses';
  const formattedExpensesTotal = numeral(expensesTotal / 100).format('$0,0.00');

  return (
    <div className='page-header'>
      <div className='page-header__content content-container'>
        <h1 className='page-header__title'>Viewing <span>{expensesCount}</span> {expenseWord} totalling <span>{formattedExpensesTotal}</span></h1>
        <Link className='button button--thin' to='/create'>Add Expense</Link>
      </div>
    </div>
  );
};

const mapStateToProps = (state) => {
  return {
    expensesCount: getVisibleExpenses(state.expenses, state.filters).length,
    expensesTotal: getExpensesTotal(getVisibleExpenses(state.expenses, state.filters))
  };
};

export default connect(mapStateToProps)(ExpensesSummary);