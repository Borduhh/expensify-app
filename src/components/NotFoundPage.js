import React from 'react';
import { Link } from 'react-router-dom';

const NotFoundPage = () => (
  <div>
    This is the Not Found Page component.
    <Link to='/'>Homepage</Link>
  </div>
);

export default NotFoundPage;