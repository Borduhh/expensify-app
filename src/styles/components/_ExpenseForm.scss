.expense-form {
  .input-group {
      background:#444;
      
    .input-group__item {
      width:49%;  
    }
  }
}