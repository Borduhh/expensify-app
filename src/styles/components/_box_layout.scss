.box-layout {
  align-items: center;
  background: url('/images/bg.jpg');
  
  background-size: cover;
  display:flex;
  height: 100vh;
  justify-content: center;
  width: 100vw;
}

.box-layout__box {
  background: fade-out($color: #FFFFFF, $amount: 0.15);
  border-radius: 3px;
  padding: $xl-size $l-size;
  max-width: 35rem;
  text-align: center;
}

.box-layout__title {
  line-height: 1;
  margin: 0 0 $m-size 0;
}