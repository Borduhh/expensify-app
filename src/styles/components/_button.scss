.button { 
  background: $light-blue;
  border: solid $mid-blue;
  border-radius: 2px;
  border-width: 0 0 5px;
  color: $white;
  font-size: $fs-large;
  font-weight: 200;
  padding: $m-size;
  text-decoration: none;

  transition: $trans-what-all $trans-time-normal $trans-how-ease;
  -webkit-transition: $trans-what-all $trans-time-normal $trans-how-ease;
  -moz-transition: $trans-what-all $trans-time-normal $trans-how-ease;   
  -ms-transition: $trans-what-all $trans-time-normal $trans-how-ease;   
  -o-transition: $trans-what-all $trans-time-normal $trans-how-ease;

  &:hover {
    background: $mid-blue;
    border-color: $dark-blue;
  }
}

.button--thin {
  border-width: 0 0 3px;
  font-size: $fs-medium;
  padding: $xs-size $l-size;
}

.button--link {
  background:none;
  border:none;
  padding: 0;
}