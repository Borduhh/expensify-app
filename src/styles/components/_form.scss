.form {
  display: flex;
  flex-direction: column;

  >.text-input, .select, .textarea {
    margin-bottom: $m-size;
  }
}

.form__error {
  font-style: italic;
  margin: 0 0 $m-size;
}