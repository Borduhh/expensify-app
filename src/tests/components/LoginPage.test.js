import { shallow } from 'enzyme';
import React from 'react';

import { LoginPage } from '../../components/LoginPage';
import startLogin from '../../actions/auth';

let startLoginSpy, wrapper;

beforeEach(() => {
startLoginSpy = jest.fn();

  wrapper = shallow(<LoginPage 
    startLogin={startLoginSpy}
  />);  
});

test('Should render LoginPage componenet correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

test('Should call startLogin on button click', () => {
  wrapper.find('button').simulate('click');

  expect(startLoginSpy).toHaveBeenCalled();
});