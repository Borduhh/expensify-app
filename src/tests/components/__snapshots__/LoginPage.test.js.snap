// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render LoginPage componenet correctly 1`] = `
<div
  className="box-layout"
>
  <div
    className="box-layout__box"
  >
    <h1
      className="box-layout__title"
    >
      Expensify
    </h1>
    <p>
      An expenses application that showcases React.js with Redux, Node.js, and Express.js.
    </p>
    <button
      className="box-layout__button button"
      onClick={[MockFunction]}
    >
      Login with Google
    </button>
  </div>
</div>
`;
